### PR TITLE
Add cognito v2 pretokengeneration event

### PR DESCRIPTION
**Pull request by XaaXaaX**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-token-generation.html
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/aws-lambda/trigger/cognito-user-pool-trigger/pre-token-generation.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4f9218a5a74b63b0b739bf881b8e39c0e2254fb4
</p>

File: types/aws-lambda/trigger/cognito-user-pool-trigger/pre-token-generation.d.ts [hunk 1]
```diff
@@ -51,3 +51,67 @@ export type PreTokenGenerationTriggerEvent =
     | PreTokenGenerationRefreshTokensTriggerEvent;
 
 export type PreTokenGenerationTriggerHandler = Handler<PreTokenGenerationTriggerEvent>;
+
+
+/** 
+ * @see https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-token-generation.html#cognito-user-pools-lambda-trigger-syntax-pre-token-generation
+*/
+export interface IdTokenGenration {
```
Comment (simonbuchan) [inline types/aws-lambda/trigger/cognito-user-pool-trigger/pre-token-generation.d.ts:59]: Typo: missing e after "Gen"

**Review by simonbuchan — APPROVED** [review]
The v2 should probably be broken out to it's own trigger file, but otherwise this looks good! (Bar that typo)

File: types/aws-lambda/trigger/cognito-user-pool-trigger/pre-token-generation.d.ts (hunk 1 above)
Comment (XaaXaaX) [reply types/aws-lambda/trigger/cognito-user-pool-trigger/pre-token-generation.d.ts]: done , thanks for attention

**Review by simonbuchan — APPROVED** [review]

**Review by jamesorlakin — APPROVED** [review]
Looks good, but I agree perhaps it could be split into a v2 file

**Comment by XaaXaaX** [comment]
> Looks good, but I agree perhaps it could be split into a v2 file

I ll do the changes , initially i was on the same plan, but looking at apigateway models i got the mix and i followed same path ;) 

i ll do the separation

**Comment by XaaXaaX** [comment]
The PreTokenGeneration V2 Event was moved to a new definition file per @simonbuchan advice

**Review by simonbuchan — APPROVED** [review]
Thanks, appreciate the effort!

**Review by darbio — APPROVED** [review]

**Review by jamesorlakin — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]

**Comment by XaaXaaX** [comment]
Ready to merge

**Pull request merged**